cpu/msp430_common/include/irq_arch.h

```diff
@@ -38,13 +45,7 @@ __attribute__((always_inline)) static inline unsigned int irq_disable(void)
     __asm__ volatile(
         "mov.w r2, %[state]"                "\n\t"
         "bic %[gie], r2"                    "\n\t"
-        /*
-         * BEWARE: IRQs remain enabled for one instruction after clearing the
-         * GIE bit in the status register (r2). Thus, the next instruction is
-         * not only used to sanitize the IRQ state, but also delays the actual
-         * critical section by one CPU cycle, so that IRQs are indeed disabled
-         * by then.
-         */
+        "nop"                               "\n\t"
```

Comment:
@maribu I think this was updated by you lately? gcc 9 complained that there might be nops missing. I agree with your reasoning that they shouldn't be necessary, but decided that I'd rather have gcc happy than convinced otherwise (by disabling the warnings with pragmas). Is that fine with you?

Reply:
Generally I agree with just making GCC happy, as it is simpler and more readable than having the explanation and the pragmas. But as `irq_disable()` is quite a hot code path in RIOT's code base, I would personally go for extra effort here.

Reply:
ok. I'll look into that once this passes all tests.

Reply:
I'd like to postpone this for a follow up, would that be alright?